vunit/sim_if/ghdl.py

```diff
@@ -364,3 +381,67 @@ def simulate(  # pylint: disable=too-many-locals
             subprocess.call(cmd)
 
         return status
+
+    def _compile_source_file(self, source_file, printer):
+        """
+        Runs parent command for compilation, and moves the .gcno file (if any) to the compilation output folder
+        """
+        compilation_ok = super()._compile_source_file(source_file, printer)
+
+        # GCOV gcno files are output to where the command is run,
+        # move it back to the compilation folder
+        source_path = Path(source_file.name)
+        gcno_file = Path(source_path.stem + ".gcno")
+        if Path(gcno_file).exists():
+            new_path = Path(source_file.library.directory) / gcno_file
+            gcno_file.rename(new_path)
+
+        return compilation_ok
+
+    def merge_coverage(self, file_name, args=None):
+        """
+        Merge coverage from all test cases
+        """
+
+        dir_name = file_name
+
+        # Loop over each .gcda output folder and merge them two at a time
+        first_input = True
+        for coverage_dir in self._coverage_dirs:
+            if file_exists(coverage_dir):
+                if first_input:
+                    # There is no output to merge with, so merge with itself to produce first output
+                    first_dir = coverage_dir
+                else:
+                    first_dir = dir_name
+                merge_command = [
+                    "gcov-tool",
+                    "merge",
+                    "-o",
+                    dir_name,
+                    first_dir,
+                    coverage_dir,
+                ]
+                subprocess.call(merge_command)
+                first_input = False
+            else:
+                LOGGER.warning("Missing coverage directory: %s", coverage_dir)
+
+        # Find actual output path of the .gcda files (they are deep in hierarchy)
+        dir_path = Path(dir_name)
+        gcda_files = dir_path.glob("**/*.gcda")
+        gcda_dirs = [x.parent for x in gcda_files]
+        gcda_dirs = set(gcda_dirs)
+        assert len(gcda_dirs) == 1, "Expected exactly one folder with gcda files"
+        gcda_dir = str(gcda_dirs.pop())
+
+        # Add compile-time .gcno files as well, they are needed or the report
+        if file_exists(dir_name):
+            for library in self._project.get_libraries():
+                lib_dir_path = Path(library.directory)
+                if lib_dir_path.exists():
+                    gcno_files = lib_dir_path.glob("*.gcno")
+                    for gcno_file in gcno_files:
```

Comment:
```suggestion
                for gcno_file in Path(library.directory).glob("*.gcno"):
```
The glob will return empty if `lib_dir_path` does not exist

Reply:
Nice